### ISSUE TITLE
Mistral-instruct not using system prompt. 

### ISSUE THREAD

**Issue by AkiraRy** [abetlen/llama-cpp-python#1832]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [+ ] I am running the latest code. Development is very rapid so there are no tagged versions as of now.
- [+ ] I carefully followed the [README.md](https://github.com/abetlen/llama-cpp-python/blob/main/README.md).
- [+ ] I [searched using keywords relevant to my issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/filtering-and-searching-issues-and-pull-requests) to make sure that I am creating a new issue that is not already open (or closed).
- [+ ] I reviewed the [Discussions](https://github.com/abetlen/llama-cpp-python/discussions), and have a new bug or useful enhancement to share.

# I delete everything else as it is not relevant. 

# TL;DR
Using mistral-instuct formating doesn't include system prompt. I don't know if this was intential or just missed out. 
Here is relevant part of code 
```python
@register_chat_format("mistral-instruct")
def format_mistral_instruct(
    messages: List[llama_types.ChatCompletionRequestMessage],
    **kwargs: Any,
) -> ChatFormatterResponse:
    eos = "</s>"
    stop = eos
    prompt = ""
    for message in messages:
        if (
            message["role"] == "user"
            and message["content"] is not None
            and isinstance(message["content"], str)
        ):
            prompt += "[INST] " + message["content"]
        elif message["role"] == "assistant" and message["content"] is not None:
            prompt += " [/INST]" + message["content"] + eos
    prompt += " [/INST]"
    return ChatFormatterResponse(prompt=prompt, stop=stop)
    
```

**Comment by ddh0** [comment]
Hi @AkiraRy, the mistral instruct prompt format doesn't officially support a system prompt, so this is probably intentional

**Comment by AkiraRy** [comment]
Oh, that's explain why `cjpais/llava-1.6-mistral-7b-gguf` doesn't really follow it's system prompt even if i define it and pass manually
